avm/ptn/deployment-script/import-image-to-acr/main.bicep

```diff
@@ -0,0 +1,255 @@
+}
+
+resource newManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = if (!useExistingManagedIdentity) {
+  name: managedIdentityName
+  location: location
+  tags: tags
+}
+
+resource existingManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (useExistingManagedIdentity) {
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = if (!empty(rbacRoleNeeded)) {
+  name: guid(acr.id, rbacRoleNeeded, useExistingManagedIdentity ? existingManagedIdentity.id : newManagedIdentity.id)
+  scope: acr
+  properties: {
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', rbacRoleNeeded)
+    principalId: useExistingManagedIdentity
+      ? existingManagedIdentity.properties.principalId
+      : newManagedIdentity.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource importImage 'Microsoft.Resources/deploymentScripts@2023-08-01' = {
+  name: 'ACR-Import-${name}-${last(split(replace(image,':','-'),'/'))}'
+  location: location
+  tags: tags
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${useExistingManagedIdentity ? existingManagedIdentity.id : newManagedIdentity.id}': {}
+    }
+  }
+  kind: 'AzureCLI'
+  dependsOn: [rbac]
+  properties: union(
+    {
+      forceUpdateTag: forceUpdateTag
+      azCliVersion: '2.59.0'
+      timeout: 'PT30M' // set timeout to 30m
+      retentionInterval: 'PT1H' // cleanup after 1h
+      environmentVariables: [
+        {
+          name: 'acrName'
+          value: acrName
+        }
+        {
+          name: 'imageName'
+          value: image
+        }
+        {
+          name: 'overwriteExistingImage'
+          value: toLower(string(overwriteExistingImage))
+        }
+        {
+          name: 'initialDelay'
+          value: '${string(initialScriptDelay)}s'
+        }
+        {
+          name: 'retryMax'
+          value: string(retryMax)
+        }
+        {
+          name: 'retrySleep'
+          value: '5s'
+        }
+      ]
+      scriptContent: '''#!/bin/bash
+set -e
+
+echo "Waiting on RBAC replication ($initialDelay)\n"
+sleep $initialDelay
+
+# retry loop to catch errors (usually RBAC delays, but 'Error copying blobs' is also not unheard of)
+retryLoopCount=0
+until [ $retryLoopCount -ge $retryMax ]
+do
+  echo "Importing Image ($retryLoopCount): $imageName into ACR: $acrName\n"
+  if [ $overwriteExistingImage = 'true' ]; then
+    az acr import -n $acrName --source $imageName --force && break
+  else
+    az acr import -n $acrName --source $imageName && break
+  fi
+
+  sleep $retrySleep
+  retryLoopCount=$((retryLoopCount+1))
+done
+
+echo "done\n"'''
+      cleanupPreference: cleanupPreference
+    },
+    storageSettings,
+    containerSettings
+  )
+
+  resource logs 'logs' existing = {
```

Comment:
needed?

Reply:
It is used in the deploymentScriptOutput output

Reply:
Fair - but - by now the 'deploymentScript' module provides an output `deploymentScriptLogs` that you can pass thru :)

Reply:
Now it is. Wasn't last time I checked. Will use it.